### PR TITLE
Migrate frontend

### PR DESCRIPTION
**Pull request by julien2429**
- [x] Check if Cypress works locally on Windows (related to #12) (Nu meeeereee)
- [x] Remove legacy Tailwind config brought from the old project
- [x] Fix Firefox bug where the search menu isn't centered
- [x] Fix inconsistent naming throughout the repository (mixed usage of snake_case and PascalCase for filenames and component names)
- [x] Fix inconsistencies with [the prototype](https://www.figma.com/file/eNTTW9Hj7KzglbLifcj8Zt/Prototip)
  - [x] fix font family, sizing, and spacing across the whole site (use the preconfigured fonts in Tailwind)
    - [x] fix fonts on work cards
      - [x] implement auto-resizing for the work card's title
      - [x] align title to bottom
  - [x] fix upload page
    - [x] remove background blur
    - [x] center content vertically
  - [x] fix upload details page
    - [x] window not scaling up from the upload method chooser from the upload page
    - [x] transition between title/characterization field not being the same as in the design
  - [x] fix search menu work card border too thin
  - [x] add work count to work cards
- [x] Encapsulate scrollbar padding style in class
- [x] Use md spacing everywhere except in the essays container, and sm spacing between rows
- [x] Make links occupy the whole grid cell
- [x] Make the upload button plus sign larger
- [x] Remove fade effect from upload details form
- [x] Use svelte-material-icons instead of own SVG files
- [x] Add bottom margin
- [x] Disallow logo text selection
- [x] Disable "Plagiat" and "Profesori" links, and disallow their selection as on the logo
- [x] Add drop shadow for the search bar, "Intră în cont" link, and for the "Eseuri" and "Caracterizări" filter buttons on the search page
- [x] Fix background for the upload details page not transitioning like the page's menu
- [x] Fix flaky underlining for:
  - [x] "Eseuri", "Caracterizări" selectors on
    - [x] index
    - [x] search
    - [x] upload (here named "Eseu", "Caracterizare")
    - [x] register (here named "Elev", "Profesor")
  Tip: underline the one that is selected, not the one in focus, so clicking outside the button does not remove its underlining.
- [x] Add custom focus outline for buttons and links
- [x] Fix Safari bug where blob glitches due to `rotateX` and `rotateY` CSS transforms
- [x] Fix search URL query string being one keystroke behind the actual search query

### CODE REVIEW

**Comment by tmaxmax** [comment]
Nu trebuie modificata configurarea tailwind, se vor folosi valorile de acolo!

File: web/src/style.css [hunk 1]
```diff
@@ -5,3 +5,11 @@
     backdrop-filter: blur(100px);
   }
 }
+:root {
+  --alb: #fcfaf9;
+  --negru: #000000;
+  --l_row: 4.133rem;
+  --l_gap_row: 1rem;
+  --l_column: 7.6rem;
+  --l_gap_column: 1.6rem;
```
Comment (tmaxmax) [inline web/src/style.css]: Nu este nevoie de astea

File: web/windi.config.ts [hunk 2]
```diff
@@ -54,6 +54,11 @@ export default defineConfig({
       sm: '0.625rem',
       md: '1.25rem',
       lg: '1.875rem',
+      xlg: '3.700rem',
+      l_row: '4.133rem',
+      l_gap_row: '1.000rem',
+      l_column: '7.555rem',
+      l_gap_column: '1.600rem',
```
Comment (tmaxmax) [inline web/windi.config.ts]: Nu este nevoie de astea

**Review by tmaxmax — CHANGES_REQUESTED** [review]

File: web/src/App.svelte [hunk 3]
```diff
@@ -1,12 +1,21 @@
...
   import { Router } from '@roxi/routify'
   import Window from '$/components/Window.svelte'
   import { routes } from '../.routify/routes'
 </script>
 
-<Router {routes} />
+<Auth0Context
+  domain="eseuri-prod.eu.auth0.com"
+  client_id="NJgHdaXoixiNUpEks7b3TtYi95ZH433P"
+  audience="https://eseuri-prod.eu.auth0.com/api/v2/"
```
Comment (tmaxmax) [inline web/src/App.svelte]: Astea nu ar trebui sa existe in repository, vezi [cum am facut eu](https://github.com/julien2429/Eseuri_fork/commit/37ee2d75089694c644d77789cd5f876e046d7b9e). Desi acolo au ramas doua dintre ele, nu ar trebui sa existe niciuna dintre datele astea, de fapt.
Citeste si [documentatia Vite](https://vitejs.dev/guide/env-and-mode.html#env-variables).

**Review by tmaxmax — CHANGES_REQUESTED** [review]

File: web/src/style.css [hunk 4]
```diff
@@ -5,6 +5,9 @@
     backdrop-filter: blur(100px);
   }
 }
+.padding {
+  padding-left: calc(100vw - 100%);
+}
```
Comment (tmaxmax) [inline web/src/style.css]: Nume mai sugestiv? Ex: `scrollbar-window-padding`

File: web/src/pages/upload/index.svelte [hunk 5]
```diff
@@ -170,9 +170,7 @@
     grid-auto-rows: var(--l_row);
     max-width: calc(var(--l_column) * 6 + var(--l_gap_column) * 5);
   }
-  .blur {
-    backdrop-filter: blur(100px);
-  }
+
   .white_thing {
     position: relative;
     margin-left: calc(-1 * (var(--l_gap_column)));
```
Comment (tmaxmax) [inline web/src/pages/upload/index.svelte]: Nume mai sugestiv?

Comment (julien2429) [reply web/src/pages/upload/index.svelte]: Este extrem de sugestiv

File: web/src/pages/index.svelte [hunk 6]
```diff
@@ -8,10 +8,10 @@
   import Buton from '$/components/buton.svelte'
   import Search from '$/components/search_bar.svelte'
   import UploadButton from '$/components/upload_button.svelte'
-  import Eseu from '$/components/eseu.svelte'
+  import Eseu from '$/components/essay.svelte'
```
Comment (tmaxmax) [inline web/src/pages/index.svelte]:
```suggestion
  import Essay from '$/components/essay.svelte'
```

File: web/src/pages/search/[page_name].svelte [hunk 7]
```diff
@@ -1,7 +1,7 @@
 <script lang="ts">
-  import Eseu from '$/components/eseu.svelte'
-  import Link from '$/components/Link.svelte'
-  import LoginButton from '$/components/LoginButton.svelte'
+  import Eseu from '$/components/essay.svelte'
```
Comment (tmaxmax) [inline web/src/pages/search/[page_name].svelte]:
```suggestion
  import Essay from '$/components/essay.svelte'

```

**Review by tmaxmax — DISMISSED** [review]
Nu trebuie pusă nicio tranziție aici

File: web/windi.config.ts [hunk 8]
```diff
@@ -51,15 +51,12 @@ export default defineConfig({
...
     },
     spacing: {
-      sm: 'var(--row-gap, 0.625rem)',
+      sm: '0.625rem',
       md: '1.25rem',
       lg: 'var(--column-gap, 1.875rem)',
       xlg: '3.700rem',
       l_row: '4.133rem',
-      l_gap_row: '0.625rem',
       l_column: '7.555rem',
```
Comment (tmaxmax) [inline web/windi.config.ts]: `l_row` și `l_column` sunt folosite undeva?

File: web/windi.config.ts [hunk 9]
```diff
@@ -51,15 +51,12 @@ export default defineConfig({
       ],
     },
     spacing: {
-      sm: 'var(--row-gap, 0.625rem)',
+      sm: '0.625rem',
       md: '1.25rem',
       lg: 'var(--column-gap, 1.875rem)',
```
Comment (tmaxmax) [inline web/windi.config.ts]: Ar trebui setate `sm` si `lg` la valorile variabilelor, iar variabilele la valorile statice respective

Comment (julien2429) [reply web/windi.config.ts]: Nu inteleg la ce te referi

File: web/windi.config.ts (hunk 8 above)
Comment (julien2429) [reply web/windi.config.ts]: La calcularea la max-width

Comment (tmaxmax) [reply web/windi.config.ts]: ma refer folosesti clasele tailwind generate pe undeva? daca nu, sterge-le din config.

File: web/windi.config.ts (hunk 9 above)
Comment (tmaxmax) [reply web/windi.config.ts]: `windi.config.ts`
```js
{
  sm: 'var(--row-gap)',
  // ...
  lg: 'var(--column-gap)'
}
```

`src/style.css`
```css
:root {
  --row-gap: 0.625rem;
  --column-gap: 1.875rem;
  /* recalculeaza valorile ce depind de variabilele modificate */
}
```

Comment (julien2429) [reply web/windi.config.ts]: Nu e md in loc de lg?

Comment (tmaxmax) [reply web/windi.config.ts]: nu, din moment ce e deja folosit la `lg`, iar `--column-gap` e folosit în calculul variabilelor pentru layout-ul eseurilor. ar trebui redenumită din `--column-gap` în `--essay-column-gap`.

**Review by tmaxmax — DISMISSED** [review]

**Review by julien2429 — COMMENTED** [review]
Foarte bine

**Review by tmaxmax — APPROVED** [review]

**Pull request merged**